src/app/common/hooks/use-location-state.ts

```diff
@@ -7,6 +7,8 @@ import { isUndefined } from '@shared/utils';
 
 export function useLocationState(propName: string): string | undefined;
 export function useLocationState(propName: string, defaultValue: string): string;
+export function useLocationState(propName: 'accountIndex'): number;
+export function useLocationState(propName: 'backgroundLocation'): Location;
```

Comment:
These changes mean `useLocationState` is now aware of implementation details of some of the cases in which it's used.

Why not extend the function to support a generic type argument, rather than extending it with use-case specific typings?

Reply:
I can change this, no problem. I struggled with extending the type initially so opted for this to narrow the type to their key. I'll clean it up